src/services/questService.ts

```
@@ -24,11 +35,31 @@
+): void => {
+    const quest = inventory.QuestKeys.find(quest => quest.ItemType === KeyChain);
+
+    if (!quest) {
+        throw new Error(`Quest ${KeyChain} not found in QuestKeys`);
+    }
+
+    if (!quest.Progress) {
+        throw new Error(`Progress should always exist when giving keychain triggered items or messages`);
+    }
+
+    let questStage = quest.Progress[ChainStage];
```

ESLint

[error] 'questStage' is never reassigned. Use 'const' instead.

(prefer-const)